cmd/tailscale/cli/web.go

```diff
@@ -94,20 +96,24 @@ func runWeb(ctx context.Context, args []string) error {
 	if prefs, err := localClient.GetPrefs(ctx); err == nil {
 		existingWebClient = prefs.RunWebClient
 	}
-	if !existingWebClient {
+	if !existingWebClient && !webArgs.readonly {
```

Comment:
I think you also have to check down below in the cleanup on `ctx.Done()` to not stop it if we were in readonly mode.

Could just add a var here `processStartedManagmentClient` or something that we can set in this `if` and just use below.